ts/components/dialog/user-settings/pages/ProxySettingsPage.tsx

```diff
@@ -0,0 +1,199 @@
+import { Flex } from '../../../basic/Flex';
+import { SessionButton, SessionButtonColor } from '../../../basic/SessionButton';
+
+const ProxyInputsContainer = styled(Flex)`
+  width: 100%;
+  gap: var(--margins-md);
+  padding: var(--margins-md) 0;
+`;
+
+type ProxySettings = {
+  enabled: boolean;
+  host: string;
+  port: string;
+  username: string;
+  password: string;
+};
+
+async function loadProxySettings(): Promise<ProxySettings> {
+  const enabled = Boolean(window.getSettingValue(SettingsKey.proxyEnabled));
+  const host = (window.getSettingValue(SettingsKey.proxyHost) as string) || '';
+  const port = String(window.getSettingValue(SettingsKey.proxyPort) || '');
+  const username = (window.getSettingValue(SettingsKey.proxyUsername) as string) || '';
+  const password = (window.getSettingValue(SettingsKey.proxyPassword) as string) || '';
+
+  return { enabled, host, port, username, password };
+}
+
+function validateProxySettings(settings: ProxySettings): { valid: boolean; error?: string } {
+  if (!settings.enabled) {
+    return { valid: true };
+  }
+
+  if (!settings.host || settings.host.trim() === '') {
+    return { valid: false, error: tr('proxyValidationErrorHost') };
+  }
+
+  const portNum = parseInt(settings.port, 10);
+  if (isNaN(portNum) || portNum < 1 || portNum > 65535) {
+    return { valid: false, error: tr('proxyValidationErrorPort') };
+  }
+
+  return { valid: true };
+}
+
+async function saveProxySettings(settings: ProxySettings): Promise<void> {
+  const validation = validateProxySettings(settings);
+  if (!validation.valid) {
+    ToastUtils.pushToastError('proxyValidationError', validation.error || '');
+    return;
+  }
+
+  await window.setSettingValue(SettingsKey.proxyEnabled, settings.enabled);
+  await window.setSettingValue(SettingsKey.proxyHost, settings.host);
+  await window.setSettingValue(SettingsKey.proxyPort, parseInt(settings.port, 10) || 0);
+  await window.setSettingValue(SettingsKey.proxyUsername, settings.username);
+  await window.setSettingValue(SettingsKey.proxyPassword, settings.password);
+
+  // Notify main process to apply proxy settings
+  ipcRenderer.send('apply-proxy-settings');
+
+  ToastUtils.pushToastSuccess('proxySaved', tr('proxySavedDescription'));
+}
+
+export function ProxySettingsPage(modalState: UserSettingsModalState) {
+  const backAction = useUserSettingsBackAction(modalState);
+  const closeAction = useUserSettingsCloseAction(modalState);
+  const title = useUserSettingsTitle(modalState);
+  const forceUpdate = useUpdate();
+
+  const [settings, setSettings] = useState<ProxySettings>({
+    enabled: false,
+    host: '',
+    port: '1080',
+    username: '',
+    password: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    void (async () => {
+      const loadedSettings = await loadProxySettings();
+      setSettings(loadedSettings);
+      setIsLoading(false);
+    })();
+  }, []);
```

Comment:
once loadProxySettings is not async, this can become this, I think 
```suggestion
  useEffect(() => {
      const loadedSettings = loadProxySettings();
      setSettings(loadedSettings);
      setIsLoading(false);
  }, []);
```